standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/PersistenceManagerProvider.java

```diff
@@ -251,11 +277,14 @@ private static PersistenceManagerFactory initPMF(Configuration conf, boolean for
     if (dsp == null) {
       pmf = JDOHelper.getPersistenceManagerFactory(dsProp);
     } else {
-      try {
+      String sourceName = forCompactor ? "objectstore-compactor" : "objectstore";
+      try (DataSourceProvider.DataSourceNameConfigurator configurator =
+               new DataSourceProvider.DataSourceNameConfigurator(conf, sourceName)) {
```

Comment:
so the name of the pool will be output to the metrics file as well?

Reply:
yes, the name acts as a prefix to the metric, such as:

> "objectstore-secondary.pool.ActiveConnections" : {
      "value" : 0
    },
    "objectstore-secondary.pool.IdleConnections" : {
      "value" : 2
    },
    "objectstore-secondary.pool.PendingConnections" : {
      "value" : 0
    },
    "objectstore-secondary.pool.TotalConnections" : {
      "value" : 2
    },
    "objectstore.pool.ActiveConnections" : {
      "value" : 0
    },
    "objectstore.pool.IdleConnections" : {
      "value" : 10
    },
    "objectstore.pool.PendingConnections" : {
      "value" : 0
    },
    "objectstore.pool.TotalConnections" : {
      "value" : 10
    }

Reply:
Thanks for the clarification. Looks great.